library/src/androidTest/java/org/xmtp/android/library/HistorySyncTest.kt

```diff
@@ -0,0 +1,173 @@
+package org.xmtp.android.library
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.xmtp.android.library.messages.PrivateKey
+import org.xmtp.android.library.messages.PrivateKeyBuilder
+import org.xmtp.android.library.messages.walletAddress
+import java.security.SecureRandom
+
+@RunWith(AndroidJUnit4::class)
+class HistorySyncTest {
+    private lateinit var alixWallet: PrivateKeyBuilder
+    private lateinit var boWallet: PrivateKeyBuilder
+    private lateinit var alix: PrivateKey
+    private lateinit var alixClient: Client
+    private lateinit var bo: PrivateKey
+    private lateinit var boClient: Client
+    private lateinit var caroWallet: PrivateKeyBuilder
+    private lateinit var caro: PrivateKey
+    private lateinit var caroClient: Client
+    private lateinit var fixtures: Fixtures
+
+    @Before
+    fun setUp() {
+        fixtures = fixtures()
+        alixWallet = fixtures.alixAccount
+        alix = fixtures.alix
+        boWallet = fixtures.boAccount
+        bo = fixtures.bo
+        caroWallet = fixtures.caroAccount
+        caro = fixtures.caro
+
+        alixClient = fixtures.alixClient
+        boClient = fixtures.boClient
+        caroClient = fixtures.caroClient
+    }
+    @Test
+    fun testSyncConsent() {
```

Comment:
Two other tests that we should write is calling sync messages and automatic syncing for consent and messages.